api/v1/verticadb_webhook.go

```diff
@@ -1810,38 +1810,34 @@ func (v *VerticaDB) checkShutdownForScaleUpOrDown(oldObj *VerticaDB, allErrs fie
 // 3 B does not have any subcluster in it that has Shutdown set to true (spec/status)
 func (v *VerticaDB) checkSClusterToBeSandboxedShutdownUnset(allErrs field.ErrorList) field.ErrorList {
 	newSclusterMap := v.GenSubclusterMap()
-	statusScluterSboxMap := v.GenSubclusterSandboxStatusMap()
 	newSclusterSboxMap := v.GenSubclusterSandboxMap()
 	newSandboxIndexMap := v.GenSandboxIndexMap()
 	newSandboxMap := v.GenSandboxMap()
 	statusSclusterIndexMap := v.GenStatusSClusterIndexMap()
 	sandboxWithError := map[string]bool{}
-	for subclusterName, sandboxName := range newSclusterSboxMap {
-		oldSandboxName, isInOldSandbox := statusScluterSboxMap[subclusterName]
-		_, foundSubcluster := newSclusterMap[subclusterName]
+	for newSC, newSB := range newSclusterSboxMap {
+		_, foundSubcluster := newSclusterMap[newSC]
 		if !foundSubcluster {
 			continue // avoid panic. this error should have been reported by other functions
 		}
 		// the subcluster to be sandboxed is either a new subcluster to be added (not found in status)
 		// or an existing subcluster (found in status), the latter of which includes two scenarios:
 		//   1 the subcluster was in a sandbox (whose name is different from current sandbox name)
 		//   2 the subcluster was not in a sandbox previously
-		if !isInOldSandbox || isInOldSandbox && oldSandboxName != sandboxName {
```

Comment:
We still need the condition `if !isInOldSandbox` to cover the case 2: the subcluster was not in a sandbox previously

Reply:
After testing, I have to put back the condition "isInOldSandbox && oldSandboxName != sandboxName".
This is what I found:
When a subcluster is moved from sandbox A to sandbox B and A is shut down, , the previous rule kicks in.
When A is not shut down but B is shut down, the new rule applies.  
Only when both A and B are shut down, there will be two different error messages: one from unsandboxing and one from sandboxing.